### PR TITLE
Replace App raps with BX

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-1311

https://github.com/rainbow-me/rainbow/assets/6284525/756ff498-6818-4f8d-a351-8a57cf25f3d6

### CODE REVIEW

File: src/raps/actions/crosschainSwap.test.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+// import { Wallet } from '@ethersproject/wallet';
```
Comment (skylarbarrera) [inline src/raps/actions/crosschainSwap.test.ts:1]: redux messing w these tests yeah?

File: src/raps/actions/crosschainSwap.ts [hunk 2]
```diff
@@ -1,176 +1,215 @@
...
-
+export const crosschainSwap = async ({
+  wallet,
+  currentRap,
+  index,
+  parameters,
+  baseNonce,
+}: ActionProps<'crosschainSwap'>): Promise<RapActionResult> => {
+  const { quote, chainId, requiresApprove } = parameters;
+  const { selectedGas, gasFeeParamsBySpeed } = gasStore.getState();
```
Comment (skylarbarrera) [inline src/raps/actions/crosschainSwap.ts:110]: this requires us to finish custom gas support before merging i think

File: src/raps/actions/crosschainSwap.ts [hunk 3]
```diff
@@ -1,176 +1,215 @@
...
-      wallet,
-    };
-
-    // @ts-ignore
     swap = await executeCrosschainSwap(swapParams);
   } catch (e) {
-    logger.sentry('Error', e);
-    const fakeError = new Error('Failed to execute swap');
-    captureException(fakeError);
+    logger.error(new RainbowError('crosschainSwap: error executeCrosschainSwap'), { message: (e as Error)?.message });
```
Comment (skylarbarrera) [inline src/raps/actions/crosschainSwap.ts:150]: nice

File: src/raps/actions/unlock.ts [hunk 4]
```diff
@@ -1,211 +1,301 @@
...
-  if (alwaysRequireApprove) return true;
-
-  const cacheKey = `${accountAddress}|${address}|${contractAddress}`.toLowerCase();
-
-  const allowance = await getRawAllowance(accountAddress, assetToUnlock, contractAddress, chainId);
-
-  logger.log('raw allowance', allowance.toString());
-  // Cache that value
-  if (!isNull(allowance)) {
-    AllowancesCache.cache[cacheKey] = allowance;
```
Comment (skylarbarrera) [inline src/raps/actions/unlock.ts:202]: TIL

File: src/raps/execute.ts [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+const waitForNodeAck = async (hash: string, provider: Signer['provider']): Promise<void> => {
+  return new Promise(async resolve => {
+    const tx = await provider?.getTransaction(hash);
+    // This means the node is aware of the tx, we're good to go
+    if ((tx && tx.blockNumber === null) || (tx && tx?.blockNumber && tx?.blockNumber > 0)) {
+      resolve();
+    } else {
+      // Wait for 1 second and try again
+      await delay(1000);
```
Comment (skylarbarrera) [inline src/raps/execute.ts:102]: this seems slow to me 🤔

**Review by skylarbarrera — CHANGES_REQUESTED** [review]
code looks great, 

do think we need to finish up the custom gas logic if we use new gas store in these existing flows before we merge this (or remove and use legacy for now)

File: src/raps/actions/crosschainSwap.test.ts (hunk 1 above)
Comment (walmat) [reply src/raps/actions/crosschainSwap.test.ts:1]: yup

File: src/raps/execute.ts (hunk 5 above)
Comment (walmat) [reply src/raps/execute.ts:102]: can bump it up, just what the BX had

**Comment by walmat** [comment]
Files to migrate off old gas logic:
- [ ] GasCard.tsx
- [ ] ConfirmExchangeButton.js
- [ ] CustomGasState.js
- [ ] FeesPanel.tsx
- [ ] FeesPanelTabs.tsx
- [ ] GasSpeedButton.js
- [ ] useENSRegistrationCosts.ts
- [ ] useMaxInputBalance.ts
- [ ] useSwapInputHandlers.ts
- [ ] useSwapRefuel.ts
- [ ] ExchangeModal.tsx
- [ ] SendConfirmationSheet.tsx
- [ ] SendSheet.js
- [ ] SignTransactionSheet.tsx
- [ ] SpeedUpAndCancelSheet.js
- [ ] MintSheet.tsx
- [ ] NFTSingleOfferSheet/index.tsx

File: src/raps/execute.ts [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+function getRapFullName<T extends RapActionTypes>(actions: RapAction<T>[]) {
+  const actionTypes = actions.map(action => action.type);
+  return actionTypes.join(' + ');
+}
+
+const delay = (ms: number) => new Promise(res => setTimeout(res, ms));
+
+const waitForNodeAck = async (hash: string, provider: Signer['provider']): Promise<void> => {
+  return new Promise(async resolve => {
+    const tx = await provider?.getTransaction(hash);
```
Comment (greg-schrammel) [inline src/raps/execute.ts:103]: why not `provider.waitForTransaction`?
https://docs.ethers.org/v5/api/providers/provider/#Provider-waitForTransaction

Comment (walmat) [reply src/raps/execute.ts:103]: We could use that and wait for 1 confirmation, however for the sake of keeping code parity between app <> bx I didn't change any of the raps (functionally).

See: https://github.com/rainbow-me/browser-extension/blob/09b452e60561671d6ae25e9dc8915cde31683920/src/core/raps/execute.ts#L109

**Review by skylarbarrera — APPROVED** [review]
could use QA but lgtm

**Pull request merged**